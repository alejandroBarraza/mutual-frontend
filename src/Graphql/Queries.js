import { gql } from '@apollo/client';

//videos queries(always get last created)
export const GETVIDEOS = gql`
    query Getvideo {
        videos(limit: 1, sort: "created_at:desc") {
            created_at
            titulo
            descripcion
            links
        }
    }
`;

//statistic query(always get last 4 created)
export const GETSTATISTICS = gql`
    query GetStatistic {
        statistics(limit: 4, sort: "created_at:asc") {
            titulo
            descripcion
            created_at
        }
    }
`;

// get download content given limit and start =="offset" as arguments
export const GETDOWNLOADS_LIMIT_START = gql`
    query GetDownloads($start: Int!, $limit: Int!) {
        downloads(start: $start, limit: $limit) {
            id
            published_at
            titulo
            imagenes {
                name
                url
            }
        }
    }
`;

export const GETDOWNLOADS_SORTED_DESC = gql`
    query GetDownloads {
        downloads(sort: "created_at:desc") {
            id
            titulo
            descripcion
            archivo {
                url
            }
        }
    }
`;

export const GETNEWS_START_LIMIT = gql`
    query getNews($limit: Int!, $start: Int!) {
        news(limit: $limit, start: $start, sort: "created_at:desc") {
            id
            titulo
            descripcion
            tiempoLectura
            imagen {
                url
            }
        }
    }
`;

// get the footer elements
export const GET_ALL_FOOTER = gql`
    query getFooter {
        footer {
            selloSeguridad
            descripcion
            instagram
            facebook
            twitter
            youtube
        }
    }
`;
