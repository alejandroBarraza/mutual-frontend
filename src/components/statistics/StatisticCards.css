.card {
    color: var(--paragraph-color);
    display: flex;
    flex-direction: column;
    align-items: center;
    flex-grow: 1;
    flex-basis: 100%;
    min-height: 140px;
    max-height: 320px;
    margin: 0.5rem 0;
    border-radius: 20px;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
}
.icon-style {
    margin-top: 1.5rem;
    color: var(--mutual-color);
    font-size: 24px;
}
.card:hover {
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
}
.card h2 {
    color: var(--mutual-color);
}
.card h2,
p {
    padding: 10px 0;
}
.card p {
    text-align: center;
}
/* small  */

@media (min-width: 576px) {
    .card {
        flex-grow: 1;
        flex-basis: 40%;
        margin: 1rem 1rem 1rem 0;
    }
}
/* medium  */

@media (min-width: 768px) {
    .card {
        min-height: 200px;
        max-height: 320px;
        flex-grow: 1;
        flex-basis: 40%;
        margin: 1rem 1rem 1rem 0;
    }
}

/* x-large  */
@media (min-width: 1200px) {
    .card {
        min-height: 180px;
        max-height: 320px;
        flex-basis: 20%;
        flex-grow: 1;
        margin: -2rem 0.2rem 3rem 0rem;
        opacity: 1;
        backdrop-filter: blur(10px);
        -webkit-backdrop-filter: blur(10px);
    }
}
