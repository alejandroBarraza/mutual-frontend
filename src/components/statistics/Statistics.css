.card-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    color: var(--title-color);
}
