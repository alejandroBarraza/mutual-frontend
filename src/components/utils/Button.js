import styled from 'styled-components';
// import { css } from 'styled-components';

// buttton options:
// mutual button = green background and with color Font, ex: <button mutual></button>
// white button(default value) = white background with white font color. ex:<button mutual></button>
//FontSize as props or || 12 px as default
export const Button = styled.button`
    padding: 0.5rem 1rem;
    font-size: ${({ fontSize }) => fontSize};
    margin: 0.3rem;
    border-radius: 20px;
    border: none;
    background-color: ${({ mutual }) => mutual && 'var(--mutual-color)'};
    transition: background-color 0.5s ease, border 0.5s ease;
    color: ${({ mutual }) => mutual && '#fff'};
    border: 1px solid var(--mutual-color);
    cursor: pointer;
    text-align: center;

    &:hover {
        background-color: ${({ mutual }) => mutual && 'var(--mutual-color-hover)'};
        border: 1px solid var(--mutual-color-hover);
    }
`;

Button.defaultProps = {
    fontSize: '12px',
    color: '#5E5E5E',
};
