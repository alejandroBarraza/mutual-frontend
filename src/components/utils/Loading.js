import React from 'react';
import CircularProgress from '@mui/material/CircularProgress';
// import '../container.css';

export const Loading = () => {
    return (
        <div className='container' style={{ textAlign: 'center' }}>
            <CircularProgress />
        </div>
    );
};
