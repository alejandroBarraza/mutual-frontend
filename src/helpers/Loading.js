import React from 'react';
import { CircularProgress } from '@material-ui/core';
import '../container.css';

export const Loading = () => {
    return (
        <div className='container' style={{ textAlign: 'center' }}>
            <CircularProgress />
        </div>
    );
};
