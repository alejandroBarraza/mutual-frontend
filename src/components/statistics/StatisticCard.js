import React from 'react';
import { GETSTATISTICS } from '../../Graphql/Queries';
import { Query } from '../utils/Query';
import './StatisticCards.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
    faHandHoldingHeart,
    faHandshake,
    faStar,
    faUserPlus,
} from '@fortawesome/free-solid-svg-icons';

export const StatisticCard = () => {
    return (
        <Query query={GETSTATISTICS} id={null}>
            {({ data: { statistics } }) => {
                return (
                    <>
                        <div className='card' key={statistics[0].titulo}>
                            <FontAwesomeIcon icon={faHandHoldingHeart} className='icon-style' />
                            <h2>{statistics[0].titulo}</h2>
                            <p>{statistics[0].descripcion}</p>
                        </div>
                        <div className='card' key={statistics[1].titulo}>
                            <FontAwesomeIcon icon={faHandshake} className='icon-style' />
                            <h2>{statistics[1].titulo}</h2>
                            <p>{statistics[1].descripcion}</p>
                        </div>
                        <div className='card' key={statistics[2].titulo}>
                            <FontAwesomeIcon icon={faStar} className='icon-style' />
                            <h2>{statistics[2].titulo}</h2>
                            <p>{statistics[2].descripcion}</p>
                        </div>
                        <div className='card' key={statistics[3].titulo}>
                            <FontAwesomeIcon icon={faUserPlus} className='icon-style' />
                            <h2>{statistics[3].titulo}</h2>
                            <p>{statistics[3].descripcion}</p>
                        </div>
                    </>
                );
            }}
        </Query>
    );
};
