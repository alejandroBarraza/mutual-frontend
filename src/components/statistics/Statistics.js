import React from 'react';
import { StatisticCard } from './StatisticCard';
import './Statistics.css';

const Statistics = () => {
    return (
        <div className='container'>
            <div className='card-container'>
                <StatisticCard />
            </div>
        </div>
    );
};
export default Statistics;
