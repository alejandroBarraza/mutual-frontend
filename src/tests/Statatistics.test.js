import { render, cleanup, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { GETSTATISTICS } from '../Graphql/Queries';
import { Statistics } from '../components/statistics/Statistics';
import { Loading } from '../components/utils/Loading';

afterEach(cleanup);
const mocks = [
    {
        request: {
            query: GETSTATISTICS,
            variables: { id: null },
        },
        result: {
            data: {
                statistics: [
                    {
                        titulo: '86%',
                        descripcion: 'de Satisfacción en la atención de pacientes',
                        created_at: '2021-10-06T16:13:06.203Z',
                    },
                    {
                        titulo: '54 años',
                        descripcion: 'de Experiencia',
                        created_at: '2021-10-06T16:13:31.594Z',
                    },
                    {
                        titulo: '97.000',
                        descripcion: 'Empresas adheridas',
                        created_at: '2021-10-06T16:13:43.484Z',
                    },
                    {
                        titulo: '2.100.000',
                        descripcion: 'Trabajadores afiliados',
                        created_at: '2021-10-06T16:13:57.339Z',
                    },
                ],
            },
        },
    },
];
const mocksError = [
    {
        request: {
            query: GETSTATISTICS,
            variables: { id: null },
        },
        error: new Error('An error occurred'),
    },
];

describe('Testing <Statistics component />', () => {
    it('should render <Loading/> before graphql promise get resolved.', () => {
        const { container } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Statistics />
            </MockedProvider>
        );

        //select loading component root class.
        const loading = container.querySelector('.MuiCircularProgress-root');
        expect(loading).toBeInTheDocument();
    }),
        it('should match <Statistic/> component with Snaphshot', async () => {
            const { container } = render(
                <MockedProvider mocks={mocks} addTypename={false}>
                    <Statistics />
                </MockedProvider>
            );
            await waitFor(() => new Promise((res) => setTimeout(res, 0)));
            expect(container).toMatchSnapshot();
        });

    it('should render the Error ui component in Error Fetch Data', async () => {
        const { container } = render(
            <MockedProvider mocks={mocksError} addTypename={false}>
                <Statistics />
            </MockedProvider>
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        const error = container.querySelector('.container-error');
        console.log(error.innerHTML);
        expect(error).toMatchSnapshot();
    });
});
