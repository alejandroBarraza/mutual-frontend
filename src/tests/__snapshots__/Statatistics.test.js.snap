// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Statistics component /> should match <Statistic/> component with Snaphshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card-container"
    >
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-hand-holding-heart fa-w-18 icon-style"
          data-icon="hand-holding-heart"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M275.3 250.5c7 7.4 18.4 7.4 25.5 0l108.9-114.2c31.6-33.2 29.8-88.2-5.6-118.8-30.8-26.7-76.7-21.9-104.9 7.7L288 36.9l-11.1-11.6C248.7-4.4 202.8-9.2 172 17.5c-35.3 30.6-37.2 85.6-5.6 118.8l108.9 114.2zm290 77.6c-11.8-10.7-30.2-10-42.6 0L430.3 402c-11.3 9.1-25.4 14-40 14H272c-8.8 0-16-7.2-16-16s7.2-16 16-16h78.3c15.9 0 30.7-10.9 33.3-26.6 3.3-20-12.1-37.4-31.6-37.4H192c-27 0-53.1 9.3-74.1 26.3L71.4 384H16c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16h356.8c14.5 0 28.6-4.9 40-14L564 377c15.2-12.1 16.4-35.3 1.3-48.9z"
            fill="currentColor"
          />
        </svg>
        <h2>
          86%
        </h2>
        <p>
          de Satisfacción en la atención de pacientes
        </p>
      </div>
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-handshake fa-w-20 icon-style"
          data-icon="handshake"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M434.7 64h-85.9c-8 0-15.7 3-21.6 8.4l-98.3 90c-.1.1-.2.3-.3.4-16.6 15.6-16.3 40.5-2.1 56 12.7 13.9 39.4 17.6 56.1 2.7.1-.1.3-.1.4-.2l79.9-73.2c6.5-5.9 16.7-5.5 22.6 1 6 6.5 5.5 16.6-1 22.6l-26.1 23.9L504 313.8c2.9 2.4 5.5 5 7.9 7.7V128l-54.6-54.6c-5.9-6-14.1-9.4-22.6-9.4zM544 128.2v223.9c0 17.7 14.3 32 32 32h64V128.2h-96zm48 223.9c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zM0 384h64c17.7 0 32-14.3 32-32V128.2H0V384zm48-63.9c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16c0-8.9 7.2-16 16-16zm435.9 18.6L334.6 217.5l-30 27.5c-29.7 27.1-75.2 24.5-101.7-4.4-26.9-29.4-24.8-74.9 4.4-101.7L289.1 64h-83.8c-8.5 0-16.6 3.4-22.6 9.4L128 128v223.9h18.3l90.5 81.9c27.4 22.3 67.7 18.1 90-9.3l.2-.2 17.9 15.5c15.9 13 39.4 10.5 52.3-5.4l31.4-38.6 5.4 4.4c13.7 11.1 33.9 9.1 45-4.7l9.5-11.7c11.2-13.8 9.1-33.9-4.6-45.1z"
            fill="currentColor"
          />
        </svg>
        <h2>
          54 años
        </h2>
        <p>
          de Experiencia
        </p>
      </div>
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star fa-w-18 icon-style"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
          />
        </svg>
        <h2>
          97.000
        </h2>
        <p>
          Empresas adheridas
        </p>
      </div>
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user-plus fa-w-20 icon-style"
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
          />
        </svg>
        <h2>
          2.100.000
        </h2>
        <p>
          Trabajadores afiliados
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing <Statistics component /> should render the Error ui component in Error Fetch Data 1`] = `
<div
  class="container-error"
>
  <p>
    "An error occurred"
  </p>
</div>
`;

exports[`Testing <Statistics component /> snpahshot statistic properties graphql with data. 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card-container"
    >
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-hand-holding-heart fa-w-18 icon-style"
          data-icon="hand-holding-heart"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M275.3 250.5c7 7.4 18.4 7.4 25.5 0l108.9-114.2c31.6-33.2 29.8-88.2-5.6-118.8-30.8-26.7-76.7-21.9-104.9 7.7L288 36.9l-11.1-11.6C248.7-4.4 202.8-9.2 172 17.5c-35.3 30.6-37.2 85.6-5.6 118.8l108.9 114.2zm290 77.6c-11.8-10.7-30.2-10-42.6 0L430.3 402c-11.3 9.1-25.4 14-40 14H272c-8.8 0-16-7.2-16-16s7.2-16 16-16h78.3c15.9 0 30.7-10.9 33.3-26.6 3.3-20-12.1-37.4-31.6-37.4H192c-27 0-53.1 9.3-74.1 26.3L71.4 384H16c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16h356.8c14.5 0 28.6-4.9 40-14L564 377c15.2-12.1 16.4-35.3 1.3-48.9z"
            fill="currentColor"
          />
        </svg>
        <h2>
          86%
        </h2>
        <p>
          de Satisfacción en la atención de pacientes
        </p>
      </div>
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-handshake fa-w-20 icon-style"
          data-icon="handshake"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M434.7 64h-85.9c-8 0-15.7 3-21.6 8.4l-98.3 90c-.1.1-.2.3-.3.4-16.6 15.6-16.3 40.5-2.1 56 12.7 13.9 39.4 17.6 56.1 2.7.1-.1.3-.1.4-.2l79.9-73.2c6.5-5.9 16.7-5.5 22.6 1 6 6.5 5.5 16.6-1 22.6l-26.1 23.9L504 313.8c2.9 2.4 5.5 5 7.9 7.7V128l-54.6-54.6c-5.9-6-14.1-9.4-22.6-9.4zM544 128.2v223.9c0 17.7 14.3 32 32 32h64V128.2h-96zm48 223.9c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zM0 384h64c17.7 0 32-14.3 32-32V128.2H0V384zm48-63.9c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16c0-8.9 7.2-16 16-16zm435.9 18.6L334.6 217.5l-30 27.5c-29.7 27.1-75.2 24.5-101.7-4.4-26.9-29.4-24.8-74.9 4.4-101.7L289.1 64h-83.8c-8.5 0-16.6 3.4-22.6 9.4L128 128v223.9h18.3l90.5 81.9c27.4 22.3 67.7 18.1 90-9.3l.2-.2 17.9 15.5c15.9 13 39.4 10.5 52.3-5.4l31.4-38.6 5.4 4.4c13.7 11.1 33.9 9.1 45-4.7l9.5-11.7c11.2-13.8 9.1-33.9-4.6-45.1z"
            fill="currentColor"
          />
        </svg>
        <h2>
          54 años
        </h2>
        <p>
          de Experiencia
        </p>
      </div>
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-star fa-w-18 icon-style"
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
          />
        </svg>
        <h2>
          97.000
        </h2>
        <p>
          Empresas adheridas
        </p>
      </div>
      <div
        class="card"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user-plus fa-w-20 icon-style"
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
          />
        </svg>
        <h2>
          2.100.000
        </h2>
        <p>
          Trabajadores afiliados
        </p>
      </div>
    </div>
  </div>
</div>
`;
